src/ert_storage/endpoints/records.py

```diff
@@ -569,26 +596,22 @@ async def _get_record_data(
 
             return Response(
                 content=stream.getvalue(),
-                media_type="application/x-numpy",
+                media_type=accept,
             )
         if accept == "text/csv":
-            data = pd.DataFrame(content)
-            labels = record.f64_matrix.labels
-            if labels is not None and realization_index is None:
-                data.columns = labels[0]
-                data.index = labels[1]
-            elif labels is not None and realization_index is not None:
-                # The output is such that rows are realizations. Because
-                # `content` is a 1d list in this case, it treats each element as
-                # its own row. We transpose the data so that all of the data
-                # falls on the same row.
-                data = data.T
-                data.columns = labels[0]
-                data.index = [realization_index]
+            data = _get_dataframe(content, record, realization_index)
 
             return Response(
                 content=data.to_csv().encode(),
-                media_type="text/csv",
+                media_type=accept,
+            )
+        if accept == "application/x-parquet":
+            data = _get_dataframe(content, record, realization_index)
+            stream = io.BytesIO()
+            data.to_parquet(stream)
+            return Response(
+                content=stream.getvalue(),
```

Comment:
`to_parquet` returns a `bytes` if you don't specify stream.

```suggestion
            return Response(
                content=data.to_parquet(),
```

Reply:
E       TypeError: to_parquet() missing 1 required positional argument: 'path'

Reply:
feature is added after 1.2.0 which is not available for py3.6